android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -184,14 +195,70 @@ protected void dispatchOnDisappear() {
         .getEventDispatcher()
         .dispatchEvent(new ScreenDisappearEvent(mScreenView.getId()));
 
+    dispatchEventInChildContainers(ScreenLifecycleEvent.Disappear);
+
+    dispatchTransitionProgress(1.0f, true, true);
+  }
+
+  public void incrementCoalescingKey() {
+    mCoalescingKey++;
+  }
+
+  protected void dispatchTransitionProgress(float alpha, boolean closing, boolean shouldIncrementCoalescingKey) {
+    if (shouldIncrementCoalescingKey) {
```

Comment:
What's the purpose of incrementing the key?

Reply:
I think we want the progress of `0.0` and `1.0` to be always sent to the JS, and such values are dispatched in lifecycle events, so we increment the key in those methods.